### ISSUE TITLE
command passed on the command line is not executed

### ISSUE THREAD

**Issue by matclab** [kovidgoyal/kitty#2312]
First I must say that I use kitty all day long and it is very nice !

Some context info:
```
$ kitty --version
kitty 0.15.1 created by Kovid Goyal
$ uname -a    
Linux xxx 5.4.13-arch1-1 #1 SMP PREEMPT Fri, 17 Jan 2020 23:09:54 +0000 x86_64 GNU/Linux
```
I tried:
```
kitty ls
kitty vim
kitty bash
```
But everytime, kitty just launch my default zsh shell…

**Comment by ctrlcctrlv** [comment]
You are doing it wrong. You must execute `kitty` as e.g.:

```bash
kitty -- vim
kitty -- bash -c "ls && read -n 1"
```

**Comment by kovidgoyal** [comment]
works fine for me. What is kitty? a wrapper script, alias? The kitty
binary??

**Comment by ctrlcctrlv** [comment]
I am now questioning my own sanity, I have no idea where I got the `--` syntax from, (perhaps git,) but it seems to work for me. :-)

But so do the examples @matclab gave. I wasn't really feeling 100% today so I think this is my cue to log off.

**Comment by kovidgoyal** [comment]
-- is a typical GNU-ism for indicating that argument processing has ended. kitty supports it but does not require it. Instead the first non-option argument automatically ends argument processing and is assumed to be the command to execute. both kitty vim and kitty -- vim will start kitty running vim inside it.

@matclab As far as I can tell this is an issue local to your machine, so am closing this bug, but feel free to post asking for more help if you need it after it has been closed.

**Comment by matclab** [comment]
Hmmm, I'll try to investigate a bit more.

In the meanwhile the following command show that kitty is the binary whose version is shown here before:
```
❯ which kitty     
/usr/bin/kitty
❯ file /usr/bin/kitty                
/usr/bin/kitty: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=c862751c4d61f7058b16044f821f9bc7878993f8, for GNU/Linux 3.2.0, stripped
```

**Comment by ctrlcctrlv** [comment]
Could `zsh` be mangling the command line?

What if you run `bash`, then `kitty` inside `bash` as `kitty bash`?

**Comment by kovidgoyal** [comment]
The commands passed to kitty are not executed via a shell, bu directly via exec()

**Comment by ctrlcctrlv** [comment]
Right, I meant "mangling the command line it sends kitty"

This can happen if there are aliases in place, or (presumably) for many other reasons besides

**Comment by matclab** [comment]
I did some debugging and it looks like the command is not executed when there is a default_session. If I remove `~/.config/kitty/default.session` it works as expected.

Is this intended ?

**Comment by kovidgoyal** [comment]
Yes, if you set startup_session in kitty.conf,
then it will always be used, as its docs state.

**Comment by matclab** [comment]
Yes indeed, I should have read deeper (to be honest I had totally forgot this startup_session stuff I've set up long ago). I feel a bid bad to have bother you with this stuff.
In fact, I did set up this `startup_session` only for setting the default layout tot `tall` for all new windows.

Is there a way to set the default layout without using a session file ?
Or is `enabled_layouts` order significant ?
